### PR TITLE
add dh orchestration

### PR DESCRIPTION
**Pull request by Renetnielsen**
## References

- #0000

### CODE REVIEW

File: .github/workflows/detect-changes.yml [hunk 1]
```diff
@@ -12,33 +12,28 @@
...
   changes:
     name: Determine relevant jobs
     runs-on: ubuntu-latest
     # Set job outputs to values from filter step
     outputs:
-      dotnet: ${{ steps.filter.outputs.dotnet }}
-      frontend: ${{ steps.affected.outputs.is-affected }}
+      dh_dotnet: ${{ steps.filter.outputs.dh_dotnet }}
+      dh_frontend: ${{ steps.dh_affected.outputs.is-affected }}
+      watt_frontend: ${{ steps.watt_affected.outputs.is-affected }}
```
Comment (djorgensendk) [inline .github/workflows/detect-changes.yml:35]: won't these two not always evaluate with the same result? If that's the case, should it be one output instead?

File: .github/workflows/detect-changes.yml [hunk 2]
```diff
@@ -82,9 +77,16 @@ jobs:
       - name: Install package dependencies
         run: yarn install --frozen-lockfile --non-interactive
```
Comment (djorgensendk) [inline .github/workflows/detect-changes.yml:78]: this step and 'PR: Fetch base branch Git history' could that be moved into the ./.github/actions/affected-action?

File: .github/workflows/dh-cd.yml [hunk 3]
```diff
@@ -26,7 +26,7 @@ jobs:
   #
 
   changes:
-    uses: ./.github/workflows/dh-detect-changes.yml
```
Comment (djorgensendk) [inline .github/workflows/dh-cd.yml:29]: the outputs from the workflow - used in several steps below have new names

File: .github/workflows/detect-changes.yml [hunk 4]
```diff
@@ -12,33 +12,28 @@
...
 # limitations under the License.
 
-# DataHub UI Frontend and BFF
-name: DH Detect changes
-
-# DESCRIPTION:
-# This workflow will determine which categories of jobs should be executed,
-# based on detected changes.
-# It should be used by the 'dh-ci.yml' and 'dh-cd.yml' to ensure they both use
-# the same path's to detect changes.
```
Comment (djorgensendk) [inline .github/workflows/detect-changes.yml:22]: the description could be changed instead of removed

File: .github/workflows/dh-ci-frontend.yml [hunk 5]
```diff
@@ -13,11 +13,10 @@
 # limitations under the License.
 
 # DataHub UI Frontend and BFF
-name: DH CI
+name: DH CI frontend
 
-concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
-  cancel-in-progress: true
```
Comment (djorgensendk) [inline .github/workflows/dh-ci-frontend.yml:20]: so concurrency isn't needed anymore? Should the new orchestrator have this instead?

File: .github/workflows/dh-ci-dotnet.yml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+env:
+  APP_NAME: app-dh
+  IS_MAIN_BRANCH: ${{ github.ref == 'refs/heads/main' }}
+  IS_PULL_REQUEST: ${{ github.event_name == 'pull_request' }}
+  NODE_OPTIONS: --max-old-space-size=6144
+  NODE_VERSION_FILE: .nvmrc
+  # keep in sync with "engines.yarn" and "volta.yarn" in "package.json"
+  YARN_VERSION: 1.22.19
+  NX_MAX_PARALLEL: 2
```
Comment (djorgensendk) [inline .github/workflows/dh-ci-dotnet.yml:29]: why are these needed

File: .github/workflows/watt-cd.yml [hunk 7]
```diff
@@ -31,7 +31,7 @@ jobs:
   #
 
   changes:
-    uses: ./.github/workflows/watt-detect-changes.yml
+    uses: ./.github/workflows/detect-changes.yml
```
Comment (djorgensendk) [inline .github/workflows/watt-cd.yml:34]: name of outputs are changed

**Review by djorgensendk — CHANGES_REQUESTED** [review]

File: .github/workflows/detect-changes.yml (hunk 1 above)
Comment (Renetnielsen) [reply .github/workflows/detect-changes.yml:35]:
```
- name: Is DH affected?
     id: dh_affected
     uses: ./.github/actions/affected
     with:
       base: ${{ github.ref == 'refs/heads/main' && 'HEAD~1' || format('origin/{0}', env.BASE_BRANCH) }}
       project: app-dh

- name: Is Watt affected?
  id: watt_affected
  uses: ./.github/actions/affected
  with:
    base: ${{ github.ref == 'refs/heads/main' && 'HEAD~1' || format('origin/{0}', env.BASE_BRANCH) }}
    project: ui-watt
```

File: .github/workflows/detect-changes.yml (hunk 2 above)
Comment (Renetnielsen) [reply .github/workflows/detect-changes.yml:78]: Other projects is using this `is-affected` action, and i have no idea what it does, so i would rather let the frontend guys refactor this when they think its needed

File: .github/workflows/detect-changes.yml [hunk 8]
```diff
@@ -12,33 +12,33 @@
...
-# DataHub UI Frontend and BFF
-name: DH Detect changes
-
 # DESCRIPTION:
 # This workflow will determine which categories of jobs should be executed,
 # based on detected changes.
-# It should be used by the 'dh-ci.yml' and 'dh-cd.yml' to ensure they both use
+# It should be used by the 'ci-orchestrator.yml', 'watt-cd.yml' and 'dh-cd.yml' to ensure they both use
 # the same path's to detect changes.
+name: Detect changes
```
Comment (djorgensendk) [inline .github/workflows/detect-changes.yml:20]:
```suggestion
# It should be used by the 'ci-orchestrator.yml', 'watt-cd.yml' and 'dh-cd.yml' to ensure they both use the same path's to detect changes.
name: Detect changes
```
